sdk/attestation/Azure.Security.Attestation/src/Models/AttestationTokenSigningKey.cs

```diff
@@ -62,17 +66,26 @@ public AttestationTokenSigningKey(X509Certificate2 certificate)
 
         private static void VerifySignerMatchesCertificate(AsymmetricAlgorithm signer, X509Certificate2 certificate)
         {
-            if (!signer.KeyExchangeAlgorithm.StartsWith(certificate.PublicKey.Key.KeyExchangeAlgorithm, System.StringComparison.Ordinal))
+#if NET6_0_OR_GREATER
+            AsymmetricAlgorithm publicKey = (AsymmetricAlgorithm)certificate.GetRSAPublicKey() ?? certificate.GetDSAPublicKey();
+#else
+            AsymmetricAlgorithm publicKey = certificate.PublicKey.Key;
+#endif
+            if (!signer.KeyExchangeAlgorithm.StartsWith(publicKey.KeyExchangeAlgorithm, System.StringComparison.Ordinal))
             {
-                throw new ArgumentException($"Signer key algorithm {signer.SignatureAlgorithm} does not match certificate key algorithm {certificate.PublicKey.Key.SignatureAlgorithm}");
+                throw new ArgumentException($"Signer key algorithm {signer.SignatureAlgorithm} does not match certificate key algorithm {publicKey.SignatureAlgorithm}");
             }
 
             // Try to match the public key in the certificate and the signer. If the platform
             // supports the ToXmlString API, then use that since it the simplest solution and is relatively fast.
             try
             {
                 string signerKey = signer.ToXmlString(false);
+#if NET6_0_OR_GREATER
+                string certificateKey = ((AsymmetricAlgorithm)certificate.GetRSAPublicKey() ?? certificate.GetDSAPublicKey()).ToXmlString(false);
+#else
                 string certificateKey = certificate.PublicKey.Key.ToXmlString(false);
```

Comment:
See: [SYSLIB0027: PublicKey.Key is obsolete](https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0027)